components/eventing-controller/cmd/eventing-controller-nats/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	subscription "github.com/kyma-project/kyma/components/eventing-controller/reconciler/subscription-nats"
+)
+
+func main() {
+	setupLog := ctrl.Log.WithName("setup")
+
+	var metricsAddr string
+	var resyncPeriod time.Duration
+	var enableDebugLogs bool
+	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.DurationVar(&resyncPeriod, "reconcile-period", time.Minute*10, "Period between triggering of reconciling calls.")
+	flag.BoolVar(&enableDebugLogs, "enable-debug-logs", false, "Enable debug logs.")
+	flag.Parse()
+
+	scheme, err := setupScheme()
+	if err != nil {
+		setupLog.Error(err, "failed to setup scheme")
+		os.Exit(1)
+	}
+
+	ctrl.SetLogger(zap.New(zap.UseDevMode(enableDebugLogs)))
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: metricsAddr,
+		Port:               9443,
+		SyncPeriod:         &resyncPeriod,
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	if err := subscription.NewReconciler(
+		mgr.GetClient(),
+		mgr.GetCache(),
+		ctrl.Log.WithName("reconciler").WithName("Subscription"),
+		mgr.GetEventRecorderFor("eventing-controller"),
+	).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to setup the Subscription controller")
```

Comment:
`"unable to setup the NATS Subscription controller"`

Reply:
Done, again please remember that this is a scaffold controller, the log messages/placeholders are not final.

Reply:
Done, but again please remember that this is a scaffold controller, the log messages/placeholders are not final.